src/org/opendatakit/briefcase/ui/export/ExportPanel.java

```diff
@@ -16,412 +16,125 @@
-
-    datePicker.getComponentToggleCalendarButton().setText("Choose...");
-    datePicker.getComponentDateTextField().setBorder(model.getBorder());
-    datePicker.getComponentDateTextField().setMargin(model.getMargin());
-
-    return datePicker;
-  }
-
-  @Override
-  public void setEnabled(boolean enabled) {
-    super.setEnabled(enabled);
-    updateForms();
-
-    for (Component c : this.getComponents()) {
-      c.setEnabled(enabled);
-    }
-    if (enabled) {
-      // and then update the widgets based upon the transfer state
-      setActiveExportState(exportStateActive);
-    }
-  }
-
-  private void setActiveExportState(boolean active) {
-    if (active) {
-      // don't allow normal actions when we are transferring...
-      btnChooseExportDirectory.setEnabled(false);
-      btnPemFileChooseButton.setEnabled(false);
-      pickStartDate.setEnabled(false);
-      pickEndDate.setEnabled(false);
-      btnExport.setEnabled(false);
-      // reset the termination future so we can cancel activity
-      terminationFuture.reset();
-    } else {
-      btnChooseExportDirectory.setEnabled(true);
-      btnPemFileChooseButton.setEnabled(true);
-      pickStartDate.setEnabled(true);
-      pickEndDate.setEnabled(true);
-      btnExport.setEnabled(true);
-      // touch-up with real state...
-      updateExportButton();
-      // retain progress text (to display last export outcome)
-    }
-    // remember state...
-    exportStateActive = active;
-  }
-
-  void resetExport() {
-  }
-
-  private List<String> export() {
-    return tableModel.getSelectedForms().parallelStream()
+  private List<String> export(ExportConfiguration defaultConfiguration) {
+    form.disableUI();
+    terminationFuture.reset();
+    List<String> errors = forms.getSelectedForms().parallelStream()
         .peek(FormStatus::clearStatusHistory)
         .map(formStatus -> (BriefcaseFormDefinition) formStatus.getFormDefinition())
-        .flatMap(formDefinition -> this.export(formDefinition).stream())
+        .flatMap(formDefinition -> this.export(defaultConfiguration, formDefinition).stream())
         .collect(toList());
+    form.enableUI();
+    return errors;
   }
 
-  private List<String> export(BriefcaseFormDefinition formDefinition) {
+  private List<String> export(ExportConfiguration defaultConfiguration, BriefcaseFormDefinition formDefinition) {
     List<String> errors = new ArrayList<>();
-    Optional<File> pemFile = Optional.ofNullable(pemPrivateKeyFilePath.getText()).map(File::new).filter(File::exists);
+    ExportConfiguration conf = forms.getConfiguration(formDefinition).orElse(defaultConfiguration);
+    Optional<File> pemFile = conf.mapPemFile(Path::toFile).filter(File::exists);
     if ((formDefinition.isFileEncryptedForm() || formDefinition.isFieldEncryptedForm()) && !pemFile.isPresent())
       errors.add(formDefinition.getFormName() + " form requires is encrypted and you haven't defined a valid private key file location");
```

Comment:
Can we improve this language, while we’re in here? “form requires is encrypted”. Maybe “form is encrypted”?